server/src/main/java/org/apache/cassandra/sidecar/acl/IdentityToRoleCache.java

```diff
@@ -47,11 +46,18 @@ public IdentityToRoleCache(Vertx vertx,
               systemAuthDatabaseAccessor::findRoleFromIdentity,
               systemAuthDatabaseAccessor::findAllIdentityToRoles,
               sidecarConfiguration.accessControlConfiguration().permissionCacheConfiguration());
-        this.systemAuthDatabaseAccessor = systemAuthDatabaseAccessor;
     }
 
     public boolean containsKey(String identity)
     {
-        return cache != null && get(identity) != null;
+        try
+        {
+            return cache() != null && get(identity) != null;
+        }
+        catch (IllegalStateException e)
```

Comment:
nit: probably throw a dedicated exception, say `CqlPreparedStatementUnavailableException`, to avoid mute other ISE by accident.

Reply:
Named it `ScehmaUnavailableException`